vault/resource_identity_entity.go

```diff
@@ -242,28 +248,59 @@ func identityEntityIDPath(id string) string {
 }
 
 func readIdentityEntityPolicies(client *api.Client, entityID string) ([]interface{}, error) {
-	resp, err := readIdentityEntity(client, entityID)
+	resp, err := readIdentityEntity(client, entityID, false)
 	if err != nil {
 		return nil, err
 	}
-	if resp == nil {
-		return nil, fmt.Errorf("error IdentityEntity %s does not exist", entityID)
-	}
 
 	if v, ok := resp.Data["policies"]; ok && v != nil {
 		return v.([]interface{}), nil
 	}
 	return make([]interface{}, 0), nil
 }
 
-// May return nil if entity does not exist
-func readIdentityEntity(client *api.Client, entityID string) (*api.Secret, error) {
+func readIdentityEntity(client *api.Client, entityID string, retry bool) (*api.Secret, error) {
 	path := identityEntityIDPath(entityID)
-	log.Printf("[DEBUG] Reading Entity %s from %q", entityID, path)
+	log.Printf("[DEBUG] Reading Entity %q from %q", entityID, path)
+
+	return readEntity(client, path, retry)
+}
+
+func readEntity(client *api.Client, path string, retry bool) (*api.Secret, error) {
+	log.Printf("[DEBUG] Reading Entity from %q", path)
+
+	var err error
+	if retry {
+		client, err = client.Clone()
+		if err != nil {
+			return nil, err
+		}
+
+		maxRetries := client.MaxRetries()
+		if maxRetries == 0 {
+			maxRetries = DefaultMaxHTTPRetries
+		}
+
+		client.SetMaxRetries(getMaxGETRetries(maxRetries))
+		client.SetCheckRetry(util.StatusCheckRetry(http.StatusNotFound))
+	}
 
 	resp, err := client.Logical().Read(path)
 	if err != nil {
-		return resp, fmt.Errorf("failed reading IdentityEntity %s from %s", entityID, path)
+		return resp, fmt.Errorf("failed reading %q", path)
 	}
+
+	if resp == nil {
+		return nil, fmt.Errorf("%s: %q", entityNotFoundError, path)
```

Comment:
Any reason you can't use %w and Is/As instead of string matching?

Reply:
Addressed in https://github.com/hashicorp/terraform-provider-vault/pull/1263/commits/aa3609cd73f8bdbd52487a84c9a735bb7e7dc729